pkg/ebpf/dns_snooper.go

```diff
@@ -101,7 +101,7 @@ func (s *SocketFilterSnooper) Close() {
 func (s *SocketFilterSnooper) run(packets <-chan gopacket.Packet) {
 	for packet := range packets {
 		layer := packet.Layer(layers.LayerTypeDNS)
-		if layer.LayerType() != layers.LayerTypeDNS {
+		if layer == nil {
```

Comment:
Curious about the previous condition got removed here, so it's no longer valid?

Reply:
The previous conditional wouldn't help us much. `packet.Layer(layers.LayerTypeDNS)` returns either `nil` or a packet for which `layer.LayerType() == layers.LayerTypeDNS` is true.